os/board/rtl8721csm/src/component/common/mbed/targets/hal/rtl8721d/flash_api.c

```diff
@@ -358,7 +358,7 @@ int  flash_stream_write(flash_t *obj, u32 address, u32 len, u8 * data)
 	u32 addr_end = (page_cnt == 1) ? (address + len) : (page_begin + 0x100);
 	u32 size = addr_end - addr_begin;
 	u8 *buffer = data;
-	u8 write_data[12];
+	u8 write_data[256];

```

Comment:
hmm here we already know pointer of buffer from above layer, I mean data, and length too.
Can't we use data directly without this extra buf? also it calls each of function based on remain data as below right?
```
size_t written_bytes = 0;
while (written_bytes < len) {
	if (len > 256) {
		FLASH_TxData256B(addr_begin, length, data);
		written_bytes += 256;
	} else if (len > 12) {
		FLASH_TxData12B(addr_begin, length, data);
		written_bytes += 12;
	} else if (len < 4) {
		FLASH_TxData4B(addr_begin, length, data);
		written_bytes += 4;
	} else {
		...
	}
	
	...
}
...
return written_bytes;
```